vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -137,6 +147,34 @@ void AdsImpl::InitializeStep4(const Result result) {
   ads_serve_->DownloadCatalog();
 }
 
+#if defined(OS_ANDROID)
+void AdsImpl::RemoveAllNotificationsAfterReboot() {
+  //ads notifications don't sustain reboot, so remove all
+  auto ads_shown_history = client_->GetAdsShownHistory();
+  if (!ads_shown_history.empty()) {
+    uint64_t ad_shown_timestamp = ads_shown_history.front();
+    uint64_t boot_timestamp = Time::NowInSeconds() -
+        static_cast<uint64_t>(base::SysInfo::Uptime().InSeconds());
+    if (ad_shown_timestamp <= boot_timestamp) {
+      notifications_->RemoveAll(false);
+    }
+  }
+}
+
+void AdsImpl::RemoveAllNotificationsAfterUpdate() {
+  std::string current_version_code (base::android::BuildInfo::GetInstance()->package_version_code());
```

Comment:
Platform specific code belongs in the service layer, and should not be a part of the ads library